### PR TITLE
Cache discover pollendpoint response

### PR DESCRIPTION
**Pull request by aaithal**
### Summary

Adds a cache for caching response from the `discover-poll-endpoint` API
### Implementation details

An LRU cache with a size of `1` has been added to cache the response from the API.
### Testing

- [X] Builds (`make release`)
- [X] Unit tests (`make short-test`) pass
- [X] Integration tests (`make test` and `make run-integ-tests`) pass
- [X] Functional tests (`make run-functional-tests`) pass

New tests cover the changes: Yes
### Description for the changelog

None
### Licensing

This contribution is under the terms of the Apache 2.0 License: Yes

### CODE REVIEW

File: agent/api/ecsclient/client.go [hunk 1]
```diff
@@ -11,98 +11,48 @@
...
+)
 
-	"github.com/aws/amazon-ecs-agent/agent/config"
-	"github.com/aws/amazon-ecs-agent/agent/ec2"
-	"github.com/aws/amazon-ecs-agent/agent/logger"
-	"github.com/aws/amazon-ecs-agent/agent/utils"
+const (
+	ecsMaxReasonLength    = 255
+	pollEndpointCacheSize = 1
+	pollEndpointCacheTTL  = 10 * time.Minute
```
Comment (samuelkarp) [inline agent/api/ecsclient/client.go]: Is this long enough?

**Comment by aaithal** [comment]
@samuelkarp  Functional tests are [passing](https://github.com/aaithal/amazon-ecs-agent/compare/dev...aaithal:cacheDiscoverPollendpoint).

File: agent/api/ecsclient/client.go (hunk 1 above)
Comment (aaithal) [inline agent/api/ecsclient/client.go]: My thoughts here are that anything >= `10m` should be good enough here. I can increase it to `30m`. What value do you have in your mind?

Comment (samuelkarp) [inline agent/api/ecsclient/client.go]: Somewhere between 10 and 30 sounds good to me.

**Comment by aaithal** [comment]
I've updated the cache TTL to `20m`. r? @samuelkarp @richardpen

**Pull request merged**